exp/controllers/machinepool_controller_phases.go

```diff
@@ -232,12 +232,16 @@ func (r *MachinePoolReconciler) reconcileInfrastructure(ctx context.Context, clu
 	// Call generic external reconciler.
 	infraReconcileResult, err := r.reconcileExternal(ctx, cluster, mp, &mp.Spec.Template.Spec.InfrastructureRef)
 	if err != nil {
-		if mp.Status.InfrastructureReady && strings.Contains(err.Error(), "could not find") {
-			// Infra object went missing after the machine pool was up and running
-			log.Error(err, "MachinePool infrastructure reference has been deleted after being ready, setting failure state")
-			mp.Status.FailureReason = capierrors.MachinePoolStatusErrorPtr(capierrors.InvalidConfigurationMachinePoolError)
-			mp.Status.FailureMessage = pointer.StringPtr(fmt.Sprintf("MachinePool infrastructure resource %v with name %q has been deleted after being ready",
-				mp.Spec.Template.Spec.InfrastructureRef.GroupVersionKind(), mp.Spec.Template.Spec.InfrastructureRef.Name))
+		if strings.Contains(err.Error(), "could not find") {
+			log.Error(err, "MachinePool infrastructure reference could not be found")
+			if mp.Status.InfrastructureReady {
+				// Infra object went missing after the machine pool was up and running
+				log.Error(err, "MachinePool infrastructure reference has been deleted after being ready, setting failure state")
+				mp.Status.FailureReason = capierrors.MachinePoolStatusErrorPtr(capierrors.InvalidConfigurationMachinePoolError)
+				mp.Status.FailureMessage = pointer.StringPtr(fmt.Sprintf("MachinePool infrastructure resource %v with name %q has been deleted after being ready",
+					mp.Spec.Template.Spec.InfrastructureRef.GroupVersionKind(), mp.Spec.Template.Spec.InfrastructureRef.Name))
+			}
```

Comment:
Do we want to se both failuremessage + reason, and the condition here?

Reply:
As mentioned in the issue, if the enduser/operator is waiting on the `ReadyCondition` to check the status of the object, setting this condition will make sure that the error condition is bubbled up.